domain/wollemi/service.go

```diff
@@ -62,14 +62,16 @@ func (this *Service) normalizePaths(paths []string) []string {
 		paths = []string{"..."}
 	}
 
-	var relpath string
-
-	if this.wd != this.root && strings.HasPrefix(this.wd, this.root) {
-		relpath = strings.TrimPrefix(this.wd, this.root+"/")
-
-		for i, path := range paths {
-			paths[i] = filepath.Join(relpath, strings.TrimSuffix(path, "/"))
+	for i, path := range paths {
+		if !filepath.IsAbs(path) {
+			path = filepath.Join(this.wd, path)
+		}
+		// TODO: should we check that all absolute paths live under the root before we get to here? i don't think we can
```

Comment:
I agree @marcuscaisey. Attempting to format a path outside of the root does not make sense. Could you add a check for that?

Reply:
Sorry for the delay. Thanks for the fix @marcuscaisey. This MR looks great to me.